apigw-ec2/README.md

```diff
@@ -0,0 +1,80 @@
+# Invoke cross account Step Function -> AWS Lambda Function
+## Architecture
+* Account Setup
+
+    ![Concept](./images/apigwec2.png)
+
+This sample pattern allows users to quickly deploy a REST API, Network Load Balancer and NGINX server on a EC2 machine in their own VPC. The pattern uses Amazon API Gateway with VPC Link to securely connect to the NLB which routes traffic to EC2 instances running NGINX server. By using this pattern, users can ensure that their private resources are not exposed to the public internet while still allowing secure access through API Gateway. This pattern can be useful in scenarios where there is a need to integrate with private resources, such as internal databases or applications, while also maintaining a high level of security.
+
+Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/apigw-ec2
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+* [Create two AWS accounts for cross account setup](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have, create them and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) install and configure one profiles with credentials for the accounts as below:
+    ```
+    [default]
+    ```
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [SAM](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) installed
```

Comment:
Also mention in the preresites if they need their own vpc and what other things they need to have configured in their account to be able to deploy this.